src/handlers/v1/markets/[chainId]/[marketId]/zap-supply/index.ts

```diff
@@ -0,0 +1,251 @@
+      }
+
+      // 2. new and append swap token logic
+      const swapLogic = await newSwapTokenLogic(chainId, _sourceToken, amount, _targetToken, slippage);
+      if (swapLogic !== undefined) {
+        logics.push(swapLogic);
+        targetTokenAmount = swapLogic.fields.output.amount;
+      } else {
+        targetTokenAmount = amount;
+      }
+
+      // 3. new and append compound v3 supply logic
+      const supplyLogic = await newSupplyLogic(
+        chainId,
+        marketId,
+        _targetToken,
+        targetTokenAmount,
+        targetCollateral === undefined,
+        new BigNumberJS(borrowUSD)
+      );
+      logics.push(supplyLogic);
+
+      const estimateResult = await apisdk.estimateRouterData({ chainId, account, logics });
+      approvals = estimateResult.approvals;
+      permitData = estimateResult.permitData;
+
+      // 4. calc target position
+      targetPosition = calculateTargetPosition(targetCollateral, targetTokenAmount, marketInfo);
+    }
+
+    const responseBody: GetZapSupplyQuotationResponseBody = {
+      quotation: { targetTokenAmount, currentPosition, targetPosition },
+      approvals,
+      permitData,
+      logics,
+    };
+
+    return formatJSONResponse(responseBody);
+  },
+};
+
+async function newSwapTokenLogic(
+  chainId: number,
+  sourceToken: common.Token,
+  amount: string,
+  targetToken: common.Token,
+  slippage?: number
+) {
+  if (sourceToken.is(targetToken)) {
+    return undefined;
+  }
+
+  if (sourceToken.is(targetToken.unwrapped)) {
+    const wrapQuotation = await apisdk.protocols.utility.getWrappedNativeTokenQuotation(chainId, {
+      input: {
+        token: sourceToken,
+        amount,
+      },
+      tokenOut: targetToken,
+    });
+    return apisdk.protocols.utility.newWrappedNativeTokenLogic(wrapQuotation);
+  }
+
+  const quotation = await apisdk.protocols.paraswapv5.getSwapTokenQuotation(chainId, {
+    input: { token: sourceToken, amount },
+    tokenOut: targetToken,
+    slippage,
+  });
+  return apisdk.protocols.paraswapv5.newSwapTokenLogic(quotation);
+}
+
+async function newSupplyLogic(
+  chainId: number,
+  marketId: string,
+  token: common.Token,
+  amount: string,
+  isBase: boolean,
+  borrorwUSD: BigNumberJS
+) {
+  if (isBase) {
+    if (borrorwUSD.gt(0)) {
+      throw newHttpError(400, { code: '400.6', message: 'borrowUSD is not zero' });
+    }
+    const tokenList = await apisdk.protocols.compoundv3.getSupplyBaseTokenList(chainId);
+    const cToken = tokenList[marketId][0][1];
+    const supplyBaseQuotation = await apisdk.protocols.compoundv3.getSupplyBaseQuotation(chainId, {
+      marketId,
+      input: {
+        token: token,
+        amount: amount,
+      },
+      tokenOut: cToken,
+    });
+
+    return apisdk.protocols.compoundv3.newSupplyBaseLogic({
+      input: supplyBaseQuotation.input,
+      output: supplyBaseQuotation.output,
+      marketId: supplyBaseQuotation.marketId,
+      balanceBps: common.BPS_BASE,
+    });
```

Comment:
Refer to [Protocolink-SDK example](https://github.com/dinngo/protocolink-js-sdk/blob/b485816df147e63ed0d79a59320b8c21494a6bc9/packages/api/examples/compound-v3/supply-base.ts#L64). I think it's best to reuse `getSupplyBaseQuotation` in case any changes in Protocolink.

Reply:
Why can `getSupplyBaseQuotation` prevent changes in Protocolink when `logic.supply-base.ts::build` solely uses the `marketId, input, balanceBps` fields? The same question applies to paraswap `getSwapTokenQuotation`.

I understand that the example is intended to illustrate each function's usage and result.

Reply:
turns out the `output` is not optional for the input of `newSupplyBaseLogic` even though it is not used.

REF: https://github.com/dinngo/protocolink-js-sdk/blob/beb04f283de5e282dc3332b6020766126b0e7078/packages/core/src/logic-types.ts#L41